src/tests/ftest/launch.py

```diff
@@ -564,7 +564,25 @@ def main():
     test_files = get_test_files(test_list)
 
     # Run all the tests
-    exit(run_tests(test_files, tag_filter, args))
+    status = run_tests(test_files, tag_filter, args)
+
+    # Process the avocado run return codes and only treat job and command
+    # failures as errors.
+    ret_code = 0
+    if status == 0:
+        print("All avocado tests passed!")
+    else:
+        if status & 1 == 1:
+            print("Detected one or more avocado test failures!")
```

Comment:
Need ret_code = 1?

Reply:
No, that is what this patch is fixing.  For use with ftest.sh we want launch.py to return a successful status if it is able to execute the avocado tests, regardless of whether they pass/fail/skip/cancel/etc.  We only want a bad status from launch.py if it encounters errors running the avocado commands.

Reply:
ok , LGTM